src/Lecture3.hs

```diff
@@ -122,21 +132,24 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
+  (<>) r1 r2 = Reward (Gold (gold r1 + gold r2)) (rewardSpecial r1 || rewardSpecial r2)
+    where gold = unGold . rewardGold
 
 
 instance Monoid Reward where
+  mempty = Reward (Gold 0) False
 
 
 {- | 'List1' is a list that contains at least one element.
 -}
 data List1 a = List1 a [a]
     deriving (Show, Eq)
 
--- | This should be list append.
 instance Semigroup (List1 a) where
-
+  (<>) (List1 a1 a2) (List1 b1 b2) = List1 a1 (a2 <> [b1] <> b2)
```

Comment:
You can slightly improve the performance here by using the `:` operator instead of list append `<>` or `++` 🐎 
```suggestion
  (<>) (List1 a1 a2) (List1 b1 b2) = List1 a1 (a2 <> (b1 : b2))
```